From: Volker Lendecke <vl@samba.org>
Date: Tue, 10 Mar 2009 15:08:38 +0000 (+0100)
Subject: Revert "Revert "Fix a valgrind error / segfault in dns_register_smbd()""
X-Git-Url: http://gitweb.samba.org/?p=samba.git;a=commitdiff_plain;h=8cb0d6280ecc62d03ea6fb5e522f57a80d9e0262

Revert "Revert "Fix a valgrind error / segfault in dns_register_smbd()""

This reverts commit 96c57d26ed790b3e94c935cc4faf5661ea43a80c.

It is REALLY, REALLY confusing that the 3_X_test build in the build farm
is 3.2 and NOT 3.3!
---

diff --git a/source/smbd/dnsregister.c b/source/smbd/dnsregister.c
index 6850d1a..ff9be43 100644
--- a/source/smbd/dnsregister.c
+++ b/source/smbd/dnsregister.c
@@ -108,7 +108,8 @@ void dns_register_smbd(struct dns_reg_state ** dns_state_ptr,
 	struct dns_reg_state *dns_state = *dns_state_ptr;
 
 	if (dns_state == NULL) {
-		*dns_state_ptr = dns_state = talloc(NULL, struct dns_reg_state);
+		dns_state = talloc_zero(NULL, struct dns_reg_state);
+		*dns_state_ptr = dns_state;
 		if (dns_state == NULL) {
 			return;
 		}
